python/tests/stress/arcticdb/version_store/test_mem_leaks.py

```diff
@@ -209,3 +383,201 @@ def proc_to_examine():
+    """
+    size = df.shape[0]
+    start_date = df.iloc[0].name
+    end_date = df.iloc[0].name
+
+    symbol = "test"
+    lib.write(symbol, df)
+    del df
+    gc.collect()
+
+    queries = []
+
+    queries = [query_group(), 
+                query_group_bigger_result(), 
+                query_apply(random_string(10)), 
+                query_resample(),
+                query_row_range(size),
+                query_date_range(start_date, end_date)]
+    
+    for rep in range(num_repetitions):
+            if (read_batch):
+                print ("RUN read_batch() tests")
+                read_requests = [ReadRequest(symbol=symbol, 
+                                             query_builder=query
+                                            ) for query in queries]
+                results_read = lib.read_batch(read_requests)
+                cnt = 0
+                for result in results_read:
+                    assert not result.data is None
+                    print_info(result.data, queries[cnt])
+                    cnt += 1
+                del read_requests, results_read
+            else:
+                print ("RUN read_batch() tests")
+                for q in queries:
+                    data: pd.DataFrame = lib.read(symbol, query_builder=q).data
+                    lib.read_batch
+                    print_info(data, q)
+                del data
+            gc.collect()
+
+    del lib, queries
+    gc.collect()
+    time.sleep(10)
+
+
+if MEMRAY_SUPPORTED: 
+    ##
+    ## PYTEST-MEMRAY integration is available only from ver 3.8 on 
+    ##
+
+    from pytest_memray import Stack
+
+    def is_relevant(stack: Stack) -> bool:
+        """
+        function to decide what to filter out and not to count specific stackframes
+
+        Stack class variables:
+
+        filename: str
+            The source file being executed, or "???" if unknown.
+
+        function: str
+            The function being executed, or "???" if unknown.
+
+        lineno: int
+            The line number of the executing line, or 0 if unknown.
+        """
+        for frame in stack.frames:
+            # do something to check if we need this to be added
+            # as mem leak
+            # print(f"SAMPLE >>> {frame.filename}:{frame.function}[{frame.lineno}]")
+            pass
+        return True
+
+    loc_limit="25 KB"
+    if (MACOS):
+        loc_limit = "35 KB"
+    @MEMRAY_TESTS_MARK
+    @pytest.mark.limit_leaks(location_limit=loc_limit, filter_fn=is_relevant)
+    def test_mem_leak_querybuilder_read_memray(library_with_symbol):
+        """
+            Test to capture memory leaks >= of specified number
+
+            NOTE: we could filter out not meaningful for us stackframes
+            in future if something outside of us start to leak using
+            the argument "filter_fn" - just add to the filter function
+            what we must exclude from calculation
+        """
+        (lib, df) = library_with_symbol
```

Comment:
This is an odd style. The pattern we use elsewhere is to use the `arctic_library_lmdb` fixture in this test, and then something like:
```
lib = arctic_library_lmdb
lib.write(symbol, construct_df_querybuilder_tests(size=2000000))
```

Reply:
If it interferes with `memray`, then this could be a separate function that is excluded by `is_relevant`

Reply:
nope it is not related with is_relevant ... It is related with the way to use fixtures.

In arcticdb you actually have fixtures that setup environment. But you have accepted that environment you setup is simple one thing. In many cases the environment will be more than one thing. In this case BOTH enormaous dataframe and empty library.

In python it is common and OK to return more than one value. In our case it is needed. We have to prepare both library and dataframe before the actual execution of the test which memray will instrument. Thus both things must hapen in the pre-setup and this is nice way both to happen then. Can they be separate fixtures? Yes, pytest allows that ...

But the test logic behind test fixture is different. They provide a way to do _both_ setup and cleanup afterwards. That is the reason this approach was chosen. This way the test is linked exactly with this fixture which intentionaly if not yet part of library with other fixtures. It is very specialized fixture that in future might evolve.

I hope this justifies ;eaving that fixture as is.

Reply:
Ah, I see that `filter_fn` is only an argument to `limit_leaks`, not `limit_memory`. I guess this is OK if fixtures are the only way to get memray to not track that memory then